data/tools/wesnoth/wmltools3.py

```diff
@@ -286,66 +294,126 @@ def isresource(filename):
     return ext and ext[1:] in resource_extensions
 
 def parse_macroref(start, line):
-    def handle_argument():
-        nonlocal opt_arg
-        nonlocal arg
-        nonlocal optional_args
-        nonlocal args
+    def handle_argument(buffer):
+        opt_arg = ""
+
+        arg = "".join(buffer)
+        if arg[0:1].isspace():
```

Comment:
With slicing, `[0:1]` is the same as writing `[0]` because the last index is not included.

Reply:
Unfortunately, ``[0]`` raises out of bounds error if the argument is empty, e.g. in ``{MACRO arg1()}``

Reply:
You're right: slicing, unlike indexing, actually returns an empty string instead of raising an error. I guess I forgot because I don't rely on this specific behavior that often...